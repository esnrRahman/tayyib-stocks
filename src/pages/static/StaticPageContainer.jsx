import React from 'react';
import PropTypes from 'prop-types';

import AboutUsContent from './content/AboutUsContent';
import ShariahComplianceContent from './content/ShariahComplianceContent';
import HelpfulWebsitesContent from './content/HelpfulWebsitesContent';

import StaticPageContainerHolder from './StaticPageContainerStyles';

const StaticPageContainer = (props) => {
  const getStaticContent = () => {
    switch (props.location.pathname) {
      case '/about_us':
        return (<AboutUsContent />);
      case '/shariah_compliance':
        return (<ShariahComplianceContent />);
      case '/helpful_websites':
        return (<HelpfulWebsitesContent />);
      default:
        return null;
    }
  };

  return (
    <StaticPageContainerHolder>
      {
       getStaticContent()
     }
    </StaticPageContainerHolder>
  );
};

StaticPageContainer.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }),
};

export default StaticPageContainer;
