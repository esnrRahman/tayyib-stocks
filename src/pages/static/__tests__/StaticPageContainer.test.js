import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import StaticPageContainer from '../StaticPageContainer';

describe('<StaticPageContainer />', () => {
  let wrapper;

  it('renders correctly for about_us', () => {
    const props = {
      location: {
        pathname: '/about_us',
      },
    };
    wrapper = shallow(<StaticPageContainer {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly for shariah_compliance', () => {
    const props = {
      location: {
        pathname: '/shariah_compliance',
      },
    };
    wrapper = shallow(<StaticPageContainer {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly for helpful_websites', () => {
    const props = {
      location: {
        pathname: '/helpful_websites',
      },
    };
    wrapper = shallow(<StaticPageContainer {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly for anything_else', () => {
    const props = {
      location: {
        pathname: '/anything_else',
      },
    };
    wrapper = shallow(<StaticPageContainer {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
