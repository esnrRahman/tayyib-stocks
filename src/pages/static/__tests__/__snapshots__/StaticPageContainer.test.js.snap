// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticPageContainer /> renders correctly for about_us 1`] = `
<styled.div>
  <AboutUsContent />
</styled.div>
`;

exports[`<StaticPageContainer /> renders correctly for anything_else 1`] = `<styled.div />`;

exports[`<StaticPageContainer /> renders correctly for helpful_websites 1`] = `
<styled.div>
  <HelpfulWebsitesContent />
</styled.div>
`;

exports[`<StaticPageContainer /> renders correctly for shariah_compliance 1`] = `
<styled.div>
  <ShariahComplianceContent />
</styled.div>
`;
